polus-feature-subsetting-plugin/src/main.py

```diff
@@ -0,0 +1,258 @@
+                for key2 in section_feat_dict[key1]:
+                    section_feat_dict[key1][key2] = sum(section_feat_dict[key1][key2])/len(section_feat_dict[key1][key2])
+                
+                # find planes to keep based on specified criteria
+                section_keep_planes.append(filter_planes(section_feat_dict[key1],removeDirection, percentile))
+            
+            # keep same planes within a section, across the minor grouping variable
+            section_keep_planes = list(section_keep_planes[0].union(*section_keep_planes))
+            section_keep_planes = [i for i in range(min(section_keep_planes), max(section_keep_planes)+1) if i in uniques[maj_grouping_var]]
+            keep_planes[section_id] = section_keep_planes
+        
+        # keep same number of planes across different sections
+        keep_planes = make_uniform(keep_planes, uniques[maj_grouping_var], delay)
+        
+        # start writing summary.txt
+        summary = open(os.path.join(outDir, 'metadata_files', 'summary.txt'), 'w')
+
+        logger.info('renaming subsetted data')
+
+        # reinitialize filepattern object
+        fp = filepattern.FilePattern(inpDir, pattern=filePattern)
+
+        # rename subsetted data
+        for file in fp(group_by=sub_section_variables+grouping_variables):
+            section_id = tuple([file[0][i] for i in section_variables]) if section_variables[0] else 1
+            section_keep_planes = keep_planes[section_id]
+            rename_map = {k:v for k,v in zip(keep_planes[section_id], uniques[maj_grouping_var])}
+
+            # update summary.txt with section renaming info
+            summary.write('------------------------------------------------ \n')
+            if sectionVar.strip():
+                summary.write('Section : {} \n'.format({k:file[0][k] for k in section_variables}))
+                logger.info('Renaming files from section : {} \n'.format({k:file[0][k] for k in section_variables}))
+            summary.write('\nThe following values of "{}" variable have been renamed: \n'.format(maj_grouping_var))
+            for k,v in rename_map.items():
+                summary.write('{} ---> {} \n'.format(k,v))
+            summary.write('\n Files : \n \n')
+
+            # rename and write output
+            for f in file:
+                if f[maj_grouping_var] not in keep_planes[section_id]:
+                    continue
+
+                # old and new file name
+                old_file_name = f['file'].name
+                file_name_dict = {k.upper():v for k,v in f.items() if k!='file'}
+                file_name_dict[maj_grouping_var.upper()] = rename_map[f[maj_grouping_var]]
+                new_file_name = fp.get_matching(**file_name_dict)[0]['file'].name
+
+                # if write output collection
+                if writeOutput:
+                    shutil.copy2(os.path.join(inpDir, old_file_name),os.path.join(outDir, 'images', new_file_name))
```

Comment:
Can you send me an example output file? We should consider making this a csv file so that it's easier to process if we ever wanted to do so. Again, I know I originally said txt file, but I'm rethinking that decision now that I'm looking at it.